osu.Game/Screens/Menu/MainMenu.cs

```diff
@@ -308,5 +316,76 @@ public ConfirmExitDialog(Action confirm, Action cancel)
+        {
+            [Resolved]
+            private DialogOverlay dialogOverlay { get; set; }
+
+            [Resolved]
+            private OsuGameBase osuGame { get; set; }
+
+            public StorageErrorDialog(OsuStorage storage, OsuStorageError error)
+            {
+                HeaderText = "osu! storage error";
+                Icon = FontAwesome.Solid.ExclamationTriangle;
+
+                var buttons = new List<PopupDialogButton>();
+
+                BodyText = $"osu! encountered an error when trying to use the custom storage path ('{storage.CustomStoragePath}').\n\n";
+
+                switch (error)
+                {
+                    case OsuStorageError.NotAccessible:
+                        BodyText += $"The default storage path ('{storage.DefaultStoragePath}') is currently being used because the custom storage path is not accessible.\n\n"
+                                    + "Is it on a removable device that is not currently connected?";
+
+                        buttons.AddRange(new PopupDialogButton[]
+                        {
+                            new PopupDialogOkButton
+                            {
+                                Text = "Try again",
+                                Action = () =>
+                                {
+                                    if (!storage.TryChangeToCustomStorage(out var nextError))
+                                        dialogOverlay.Push(new StorageErrorDialog(storage, nextError));
+                                }
+                            },
+                            new PopupDialogOkButton
+                            {
+                                Text = "Use the default path from now on",
+                                Action = storage.ResetCustomStoragePath
+                            },
+                            new PopupDialogCancelButton
+                            {
+                                Text = "Only use the default path for this session",
+                            },
+                        });
+                        break;
+
+                    case OsuStorageError.AccessibleButEmpty:
+                        BodyText += "The custom storage path is currently being used but is empty.\n\n"
+                                    + "Have you moved the files elsewhere?";
+
+                        // Todo: Provide the option to search for the files similar to migration.
+                        buttons.AddRange(new PopupDialogButton[]
+                        {
+                            new PopupDialogOkButton
+                            {
+                                Text = "Reset to default",
+                                Action = storage.ResetCustomStoragePath
+                            },
+                            new PopupDialogCancelButton
+                            {
+                                Text = "Keep using the custom path"
```

Comment:
```suggestion
                                Text = "Start fresh at specified location"
```

I'd also put this button above the "reset to default"